app/src/main/java/org/mozilla/fenix/compose/PrimaryAction.kt

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Button
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * Primary action button.
+ *
+ * @param text The button text to be displayed.
+ * @param icon Optional button icon to be displayed.
+ * @param onClick Invoked when the user clicks on the button.
+ */
+@Composable
+fun PrimaryAction(
+    text: String,
+    icon: Painter? = null,
+    onClick: () -> Unit,
+) {
+    Button(
+        onClick = onClick,
+        modifier = Modifier
+            .clip(RoundedCornerShape(size = 4.dp))
+            .fillMaxWidth()
+            .height(36.dp),
```

Comment:
The `.height` is unnecessary, per the documentation. By default, `Button`s apparently have a min height of 36 and a min width of 64.

Reply:
If I don't add the height parameter there is an additional margin around the button (which we didn't have in the xml implementation).
<img width="511" alt="Screen Shot 2022-05-04 at 12 39 56 PM" src="https://user-images.githubusercontent.com/29518411/166729418-fa1e002e-449e-4bd2-9ed5-8bee23fa9f0c.png">

as opposed to when I add the height
<img width="511" alt="Screen Shot 2022-05-04 at 12 40 24 PM" src="https://user-images.githubusercontent.com/29518411/166729596-f81c85ad-95d2-41f0-a4e4-59a7a9915d18.png">

Reply:
Gotcha gotcha. You could instead just remove the vertical padding instead of setting an explicit size then (via Button's `contentPadding` parameter.

Reply:
That doesn't seem to remove the margin